services/api/src/resources/task/resolvers.ts

```diff
@@ -125,7 +126,30 @@ export const getTasksByEnvironmentId: ResolverFn = async (
     queryBuilder = queryBuilder.limit(limit);
   }
 
-  return query(sqlClientPool, queryBuilder.toString());
+  const rows = await query(sqlClientPool, queryBuilder.toString())
+
+  let adminTasks = false
```

Comment:
This is repeated three times in the code. I'm thinking that maybe we introduce something like a "filters.ts" and just refactor out to that?

Something like 

```
import * as R from 'ramda';

export const Filters = {
    filterAdminTasks: async (hasPermission, rows) => {
        let adminTasks = false
        // do a check of all returned tasks for any that are admin tasks
        for (const row of rows) {
          if (row.adminOnlyView == true) {
            adminTasks = true;
            break;
          }
        }

        // if any of the tasks are admin tasks, check the user has admin permission
        if (adminTasks) {
          // we do this so we only check admin permission once, instead of on all tasks
          try {
            await hasPermission('project', 'viewAll');
          } catch (err) {
            return rows.filter((row) => row.adminOnlyView != true);
          }
        }
        // else we simply return the rows untouched
        return rows;
    }
}
```

Which can then be called like
`rows = Filters.filterAdminTasks(hasPermissions, rows);`